### PR TITLE
Correction bugs

### PR DESCRIPTION
**Pull request by CeliaBoiss**

### CODE REVIEW

**Comment by emrecaydin** [comment]
let's have a look

File: src/Controller/RequestController.php [hunk 1]
```diff
@@ -119,7 +119,6 @@ private function issetPost(): array
     public function acceptedList()
     {
         $errors = [];
-        $answerers = [];
 
         $requestManager = new RequestManager();
         $requests = $requestManager->selectAllAcceptedRequests();
```
Comment (emrecaydin) [inline src/Controller/RequestController.php:124]: good

File: src/Controller/RequestController.php [hunk 2]
```diff
@@ -130,14 +129,7 @@ public function acceptedList()
...
-            $answerers[$i] = (
-                new UserManager())->selectAnswererInfo($requests[$i]['fk_answerer_id']);
-
-            $requests[$i]['answererFirstname'] = $answerers[$i]['firstname'];
-            $requests[$i]['answererLastname'] = $answerers[$i]['lastname'];
-        }
+        $requests = self::getAnswererInfo($requests);
 
         return $this->twig->render(
             'Request/answeredRequests.html.twig',
```
Comment (emrecaydin) [inline src/Controller/RequestController.php:135]: good

File: src/Controller/RequestController.php [hunk 3]
```diff
@@ -153,7 +145,7 @@ public function acceptedListByUser()
         $answererId = [];
 
         if ($_SERVER['REQUEST_METHOD'] === 'POST' && isset($_POST['userId'])) {
-            if ($_POST['userId'] !== 'Toutes les demandes') {
+            if ($_POST['userId'] !== '-- --') {
                 $answererId = trim($_POST['userId']);
 
                 if (filter_var($answererId, FILTER_VALIDATE_INT) !== false) {
```
Comment (emrecaydin) [inline src/Controller/RequestController.php:151]: good

File: src/Controller/RequestController.php [hunk 4]
```diff
@@ -162,6 +154,8 @@ public function acceptedListByUser()
                     $requestManager = new RequestManager();
                     $requests = $requestManager->selectAllAcceptedById($answererId);
 
+                    $requests = self::getAnswererInfo($requests);
+
                     if (!isset($requests['error'])) {
                         $users = $this->selectAllUsers();
                     } else {
```
Comment (emrecaydin) [inline src/Controller/RequestController.php:161]: good

File: src/Controller/RequestController.php [hunk 5]
```diff
@@ -251,4 +245,20 @@ public function getAnswererId(): string
...
+            $answerers[$i] = (
+                new UserManager())->selectAnswererInfo($requests[$i]['fk_answerer_id']);
+
+            $requests[$i]['answererFirstname'] = $answerers[$i]['firstname'];
+            $requests[$i]['answererLastname'] = $answerers[$i]['lastname'];
+        }
+
+        return $requests;
+    }
 }
```
Comment (emrecaydin) [inline src/Controller/RequestController.php:264]: good job

**Review by emrecaydin — APPROVED** [review]

**Review by RomainClair — APPROVED** [review]

**Pull request merged**